docs/mobile/exercises/README.md

```diff
@@ -0,0 +1,284 @@
+because tables should not be displayed with more than 5 rows as a general rule.
+
+## image
+
+```yaml
+image: my-cool-plot
+```
+
+You can put image assets into your exercises! In this example, `my-cool-plot` is
+a reference to an image asset specified in the [course
+manifest](../repo-structure.md).
+
+## complex context
+
+The `context` block can be more complex than a simple string. Sometimes you need
+to introduce a table, image, multi-line code snippet, or output in the context,
+*before* the question block. This really starts to stretch the limits of what
+can be displayed on a single screen, so simple text context should always be
+strongly preferred.
+
+For example, to include an image such as a plot:
+
+```yaml
+context:
+  text: Study the plot below to answer the following question.
+  image: my-cool-plot
+```
+
+To include a table:
+
+```yaml
+context:
+  text: Study the table below to answer the following question.
+  table: |-
+    foo, bar, baz
+    1  ,2   ,3
+    4  ,5   ,6
+```
+
+## feedback
+
+The `feedback` field is the message that is displayed when a student gets the
+exercise correct. These aren't required. In fact, you should never put any
+critical pedagogical information here. If you want to put in a message of
+encouragement or jubilation however, feel free!
+
+## feedback_wrong
+
+```yaml
+feedback_wrong: "Don't forget that R is 1-indexed, not 0-indexed like Python."
+```
+
+The `feedback_wrong` field is the fallback feedback message when a particular
+distractor hasn't been supplied with one. Usually, you should be giving specific
+feedback for each individual distractor, because specific, enlightening feedback
+messages are one of the most effective ways of teaching on DataCamp for
+Mobile. If you find that the same feedback message is appropriate for multiple
+distrators, however, this field is provided for convenience.
+
+## tabs
+
+```yaml
+tabs:
+  - key: people
+    type: table
+```
+
+Tabs appear in the top right of the exercise and serve as a way to display
+datasets or files. The appropriate asset must be defined in the [course
+manifest](../repo-structure.md). A tab field has two subfields: `key`, which is
+the alias for the asset as defined in the manifest, and `type`, which specifies
+how to render the asset. Currently only `file` and `table` are supported.
+
+# Summary of available blocks
+
+| Block             | Reorder | Tap   | Select Code | Select Output | Select Table | Multiple Choice |
+|-------------------+---------+-------+-------------+---------------+--------------+-----------------|
+| context           | :ok:    | :ok:  | :ok:        | :ok:          | :ok:         | :ok:            |
+| question          | :ok:    | :ok:  | :ok:        | :ok:          | :ok:         | :ok:            |
+| key               | :req:   | :req: | :req:       | :req:         | :req:        | :req:           |
+| code              | :req:   | :req: | :req:       | :ok:          | :ok:         | :ok:            |
```

Comment:
Multiple Choice doesn't support code / output at the moment. (I haven't checked all other fields)